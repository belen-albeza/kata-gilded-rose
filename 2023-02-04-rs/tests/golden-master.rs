use gildedrose_kata::run;

#[test]
fn test_golden_master_1_day() {
    let result = run(1);
    insta::assert_yaml_snapshot!(result);
}

#[test]
fn test_golden_master_2_days() {
    let result = run(2);
    insta::assert_yaml_snapshot!(result);
}

#[test]
fn test_golden_master_5_days() {
    let result = run(5);
    insta::assert_yaml_snapshot!(result);
}

#[test]
fn test_golden_master_10_days() {
    let result = run(10);
    insta::assert_yaml_snapshot!(result);
}
