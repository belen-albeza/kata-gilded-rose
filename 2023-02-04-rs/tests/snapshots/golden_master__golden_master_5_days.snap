---
source: tests/golden-master.rs
expression: result
---
- name: +5 Dexterity Vest
  sell_in: 5
  quality: 15
- name: Aged Brie
  sell_in: -3
  quality: 8
- name: Elixir of the Mongoose
  sell_in: 0
  quality: 2
- name: "Sulfuras, Hand of Ragnaros"
  sell_in: 0
  quality: 80
- name: "Sulfuras, Hand of Ragnaros"
  sell_in: -1
  quality: 80
- name: Backstage passes to a TAFKAL80ETC concert
  sell_in: 10
  quality: 25
- name: Backstage passes to a TAFKAL80ETC concert
  sell_in: 5
  quality: 50
- name: Backstage passes to a TAFKAL80ETC concert
  sell_in: 0
  quality: 50
- name: Conjured Mana Cake
  sell_in: -2
  quality: 0

